### PR TITLE
imgtool: Remove object pool usage

### PR DESCRIPTION
**Pull request by aaronsgiles**
Object/resource pool usage was almost gone from imgtool. This PR removes the remainder. Specifically:

* The image class doesn't allocate extra_data for the module; rather, the module allocates it on its own
* Several strings are now represented as std::string in the module structure (which can no longer be cleared by memset)
* The image::malloc() function has been removed; it was only used in one place, for the OS9 bitmap allocation. Since the maximum bitmap size is only 64k, to avoid deeper changes I just changed it to a fixed size 64k array so that the malloc is no longer needed.

With this PR, and the next one to remove usage from flopimg, we should be able to remove the very legacy src\lib\util\pool.* code altogether.

### CODE REVIEW

File: src/tools/imgtool/imgtool.cpp [hunk 1]
```diff
@@ -850,19 +839,19 @@ bool imgtool_validitychecks(void)
 	{
 		features = imgtool_get_module_features(module.get());
 
-		if (!module->name)
+		if (module->name.empty())
 		{
-			util::stream_format(std::wcerr, L"imgtool module %s has null 'name'\n", wstring_from_utf8(module->name));
+			util::stream_format(std::wcerr, L"imgtool module %s has null 'name'\n", wstring_from_utf8(module->name.c_str()));
```
Comment (cuavas) [inline src/tools/imgtool/imgtool.cpp:844]: Why are you calling `.c_str()` for every call to `wstring_from_utf8`?  The prototype at src/lib/util.h:71 takes a `std::string` reference as a parameter.  (Yes, it should be changed to a string view, but that won’t stop a `std::string` being passed directly.)

File: src/tools/imgtool/library.h [hunk 2]
```diff
@@ -459,39 +458,48 @@ char *imgtool_temp_str(void);
...
-
-	uint32_t block_size;
-
-	const util::option_guide *createimage_optguide;
-	const char *createimage_optspec;
-
-	const void *extra;
+	unsigned int tracks_are_called_cylinders : 1;    /* used for hard drivers */
+	unsigned int writing_untested : 1;               /* used when we support writing, but not in main build */
+	unsigned int creation_untested : 1;              /* used when we support creation, but not in main build */
```
Comment (cuavas) [inline src/tools/imgtool/library.h:484]: Eugh, bitfields.  Is the space saving worth the access penalty?  Would it be better to just use `bool` for all of them?

File: src/tools/imgtool/modules/os9.cpp [hunk 3]
```diff
@@ -44,7 +44,7 @@ struct os9_diskinfo
 	unsigned int octal_track_density : 1;
 
 	char name[33];
-	uint8_t *allocation_bitmap;
+	uint8_t allocation_bitmap[65536];
```
Comment (cuavas) [inline src/tools/imgtool/modules/os9.cpp:47]: Does anything create an `os9_diskinfo` on the stack?  If something does, it’s probably better to make this a `vector` or `unique_ptr`.

Comment (aaronsgiles) [reply src/tools/imgtool/modules/os9.cpp:47]: I verified that nothing creates one on the stack. I totally would have used allocated storage except that the archaic architecture of imgtool allocates raw byte buffers to hold this data, so new/delete are never called. The whole allocation strategy needs a rethink longer term. This should be sufficient for now. It's only 64k.

**Comment by galibert** [comment]
Don't worry much about it, I plan to kill that code.

   OG.

**Pull request merged**